test/python/test_authentication_external_api.py

```diff
@@ -0,0 +1,205 @@
+    oidc_request_form = {
+        'grant_type': 'password',
+        'username': 'bob',
+        'password': 'bob',
+        'scope': 'openid'
+    }
+    result = requests.post(
+        'http://oidc-keycloak:8080/auth/realms/master/protocol/openid-connect/token',
+        data=oidc_request_form,
+        auth=('conjurClient', '1234')
+    )
+    result = json.loads(result.content)
+    return result['id_token']
+
+class TestExternalAuthnApi(api_config.ConfiguredTest):
+    """Class tests api functions relating to external authenticators
+
+    Separate from the Authn tests to avoid issues with changing api keys/paswords
+    """
+    def setUp(self):
+        self.api = conjur.api.AuthenticationApi(self.client)
+        self.bad_auth_api = conjur.api.AuthenticationApi(self.bad_auth_client)
+
+    def test_enable_authenticator_instance_204(self):
+        """Test case for enable_authenticator_instance 204 response
+
+        Updates the authenticators configuration
+        """
+        _, status, _ = self.api.enable_authenticator_instance_with_http_info(
+            'authn-ldap',
+            'test',
+            self.account,
+            enabled=True
+        )
+
+        self.assertEqual(status, 204)
+
+    def test_enable_authenticator_instance_401(self):
+        """Test case for enable_authenticator_instance 401 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.bad_auth_api.enable_authenticator_instance(
+                'oidc',
+                'okta',
+                self.account,
+                enabled=False
+            )
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_enable_authenticator_instance_404(self):
+        """Test case for enable_authenticator_instance 404 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.api.enable_authenticator_instance(
+                'oidc',
+                'okta',
+                self.account,
+                enabled=False
+            )
+
+        self.assertEqual(context.exception.status, 404)
+
+    def test_service_login_200(self):
+        """Test case for service_login 200 response
+
+        Login with the given authenticator
+        """
+        alice_config = api_config.get_api_config(username='alice')
+        alice_config.password = 'alice'
+        alice_client = conjur.ApiClient(alice_config)
+        alice_api = conjur.api.AuthenticationApi(alice_client)
+
+        _, status, _ = alice_api.get_api_key_via_ldap_with_http_info('test', self.account)
+
+        self.assertEqual(status, 200)
+
+    def test_service_login_401(self):
+        """Test case for service_login 401 response"""
+        with self.assertRaises(conjur.exceptions.ApiException) as context:
+            self.bad_auth_api.get_api_key_via_ldap('aws', self.account)
+
+        self.assertEqual(context.exception.status, 401)
+
+    def test_get_access_token_service_200(self):
```

Comment:
Not directly related to this PR, but these function names need to be updated to match the newest `operationId`'s in the spec. Should probably included in the issue mentioned below.

Reply:
Included a commit updating function names in this PR, figured it was small enough.